pkg/action/executor/k8s_controller.go

```diff
@@ -136,7 +136,15 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 	}
 
 	controllerOwnerReferences := c.obj.GetOwnerReferences()
-	if !c.shouldSkipOperator(c.obj) && len(controllerOwnerReferences) > 0 && (*controllerOwnerReferences[0].Controller || "ClusterServiceVersion" == controllerOwnerReferences[0].Kind) {
+	var isControlledByOwner bool
+	if len(controllerOwnerReferences) > 0 {
+		if controllerOwnerReferences[0].Controller != nil && *controllerOwnerReferences[0].Controller {
+			isControlledByOwner = true
+		} else if "ClusterServiceVersion" == controllerOwnerReferences[0].Kind {
+			isControlledByOwner = true
+		}
+	}
+	if !c.shouldSkipOperator(c.obj) && isControlledByOwner {
```

Comment:
The !c.shouldSkipOperator(c.obj) could be before line 140 since there is no need to figure out isControlledByOwner if the former is false.

Reply:
@pankajbatra1 , I've updated my change and put check logic after `!c.shouldSkipOperator(c.obj)`